src/libraries/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/SHAHashProvider.Browser.cs

```diff
@@ -62,965 +62,34 @@ public override int GetCurrentHash(Span<byte> destination)
-            // used in wasm scenarios, and as of the current release we don't make any security guarantees
-            // about our crypto primitives in wasm environments.
-            private Sha1ForNonSecretPurposes _state; // mutable struct - don't make readonly
-
-            public override void Initialize()
-            {
-                _state = default;
-                _state.Start();
-            }
-
-            public override void HashCore(byte[] partIn, int ibStart, int cbSize)
-            {
-                _state.Append(partIn.AsSpan(ibStart, cbSize));
-            }
-
-            public override byte[] HashFinal()
-            {
-                byte[] output = new byte[20];
-                _state.Finish(output);
-                return output;
-            }
-        }
-
-        // ported from https://github.com/microsoft/referencesource/blob/a48449cb48a9a693903668a71449ac719b76867c/mscorlib/system/security/cryptography/sha256managed.cs
-        private class SHA256ManagedImplementation : SHAManagedImplementationBase
-        {
-            private byte[] _buffer;
-            private long _count; // Number of bytes in the hashed message
-            private uint[] _stateSHA256;
-            private uint[] _W;
-
-            public SHA256ManagedImplementation()
-            {
-                _stateSHA256 = new uint[8];
-                _buffer = new byte[64];
-                _W = new uint[64];
-
-                InitializeState();
-            }
-
-            public override void Initialize()
+            unsafe
             {
-                InitializeState();
-
-                // Zeroize potentially sensitive information.
-                Array.Clear(_buffer, 0, _buffer.Length);
-                Array.Clear(_W, 0, _W.Length);
-            }
-
-            private void InitializeState()
-            {
-                _count = 0;
-
-                _stateSHA256[0] = 0x6a09e667;
-                _stateSHA256[1] = 0xbb67ae85;
-                _stateSHA256[2] = 0x3c6ef372;
-                _stateSHA256[3] = 0xa54ff53a;
-                _stateSHA256[4] = 0x510e527f;
-                _stateSHA256[5] = 0x9b05688c;
-                _stateSHA256[6] = 0x1f83d9ab;
-                _stateSHA256[7] = 0x5be0cd19;
-            }
-
-            /* SHA256 block update operation. Continues an SHA message-digest
-            operation, processing another message block, and updating the
-            context.
-            */
-            public override unsafe void HashCore(byte[] partIn, int ibStart, int cbSize)
-            {
-                int bufferLen;
-                int partInLen = cbSize;
-                int partInBase = ibStart;
-
-                /* Compute length of buffer */
-                bufferLen = (int)(_count & 0x3f);
-
-                /* Update number of bytes */
-                _count += partInLen;
-
-                fixed (uint* stateSHA256 = _stateSHA256)
+                fixed (byte* src = srcArray)
```

Comment:
Is the callee resilient against _src_ or _dest_ being nullptr? (This could occur if the source or destination are non-null 0-length buffers.)

Reply:
The nullptr `src` is tested. I will verify `dest`.

Reply:
I've verified this is handled properly at the API layer.